### PR TITLE
CRH-60(feat): UI navigation flow for coding test

### PR DESCRIPTION
**Pull request by cv-sudesh-patil**
- Removed General Instructions tab from header
- Added General instruction tab on generated test link navigation and handle navigation flow for online test
- Hide header in case we are on General instructions page
- Rename component and its route and removed Authguard for coding-test route

### CODE REVIEW

File: src/app/modules/shared/components/header/header.component.html [hunk 1]
```diff
@@ -27,9 +29,9 @@
               </div>
             </div>
           </li>
-          <li class="nav-item">
+          <!-- <li class="nav-item">
```
Comment (Aakansha29Choudhary) [inline src/app/modules/shared/components/header/header.component.html]: We can remove this

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -41,7 +40,7 @@ const routes: Routes = [
 ];
 
 @NgModule({
-  imports: [RouterModule.forRoot(routes)],
+  imports: [RouterModule.forRoot(routes, { useHash: true })],
```
Comment (cv-yashi) [inline src/app/app-routing.module.ts]: I guess, it's adding # in all the urls, can we confirm this, this should be for this generate link url only and not for others

File: src/app/modules/shared/components/header/header.component.html [hunk 3]
```diff
@@ -1,12 +1,12 @@
-<div>
+<div *ngIf="!authService.hideHeader">
```
Comment (cv-yashi) [inline src/app/modules/shared/components/header/header.component.html:1]: can store it in local variable of component, as we already have other used other things of library like that

File: src/app/modules/shared/components/header/header.component.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ export class HeaderComponent implements OnInit {
   isHrTeam$?: BehaviorSubject<boolean>;
   username$?: BehaviorSubject<string | null>;
 
-  constructor(private socialAuthService: SocialAuthService, private authService: AuthService, private router: Router) {}
+  constructor(private socialAuthService: SocialAuthService, public authService: AuthService, private router: Router) {}
```
Comment (cv-yashi) [inline src/app/modules/shared/components/header/header.component.ts:16]: Let's make it private again

**Review by cv-yashi — APPROVED** [review]
Minor comments, otherwise LGTM

**Review by Aakansha29Choudhary — APPROVED** [review]

File: src/app/modules/shared/components/header/header.component.html (hunk 3 above)
Comment (cv-sudesh-patil) [reply src/app/modules/shared/components/header/header.component.html:1]: Taken it as a service variable reason is when we hit test link in new tab then header component loads first and then the general instructions component loads. So, if we use local variable then we can not hide the header. That's the reason taken service variable

File: src/app/app-routing.module.ts (hunk 2 above)
Comment (cv-sudesh-patil) [reply src/app/app-routing.module.ts]: No need, will remove it

**Pull request merged**